components/combobox/combobox.test.js

```diff
@@ -11,158 +15,141 @@ const basePropsData = {
+      listeners: { ...baseListeners, ...mockListeners },
+      localVue: testContext.localVue,
+    });
+
     inputWrapper = wrapper.find('[data-qa="dt-combobox-input-wrapper"]');
     input = wrapper.find('input');
     listWrapper = wrapper.find('[data-qa="dt-combobox-list-wrapper"]');
     skeletons = wrapper.find('[data-qa="skeleton-text-body"]');
     comboboxEmptyList = wrapper.find('[data-qa="dt-combobox-empty-list"]');
   };
 
-  const _mountWrapper = () => {
-    wrapper = mount(DtCombobox, {
-      propsData,
-      attrs,
-      slots,
-      scopedSlots,
-      listeners,
-      localVue: testContext.localVue,
-    });
-  };
-
-  // Test Setup
   beforeAll(() => {
     testContext.localVue = createLocalVue();
   });
 
   beforeEach(() => {
-    propsData = basePropsData;
-    selectStub = vi.fn();
-    escapeStub = vi.fn();
-    highlightStub = vi.fn();
-    openedStub = vi.fn();
-    listeners = { select: selectStub, escape: escapeStub, highlight: highlightStub, opened: openedStub };
-    _mountWrapper();
-    _setChildWrappers();
+    updateWrapper();
   });
 
-  // Test Teardown
   afterEach(() => {
-    propsData = basePropsData;
-    slots = {};
-    scopedSlots = {};
+    mockAttrs = {};
+    mockProps = {};
+    mockSlots = {};
+    mockScopedSlots = {};
+    mockListeners = {};
+    vi.restoreAllMocks();
   });
 
   describe('Presentation Tests', () => {
-    it(
-      'should render the component',
-      () => { expect(wrapper.exists()).toBe(true); },
-    );
+    it('should render the component', () => {
+      expect(wrapper.exists()).toBe(true);
+    });
 
     describe('When a input is provided', () => {
-      // Test Setup
-      beforeEach(async () => {
-        scopedSlots = { input: '<input v-bind="props.inputProps" />' };
-        _mountWrapper();
-        _setChildWrappers();
+      beforeEach(() => {
+        mockScopedSlots = { input: '<input v-bind="props.inputProps" />' };
+
+        updateWrapper();
       });
 
-      it(
-        'should render the input wrapper',
-        () => { expect(inputWrapper.exists()).toBe(true); },
-      );
-      it('should render the input', () => { expect(input.exists()).toBe(true); });
+      it('should render the input wrapper', () => {
+        expect(inputWrapper.exists()).toBe(true);
+      });
+
+      it('should render the input', () => {
+        expect(input.exists()).toBe(true);
+      });
     });
 
     describe('When label is provided', () => {
-      beforeEach(async () => {
-        scopedSlots = { input: '<input v-bind="props.inputProps" />' };
-        _mountWrapper();
-        _setChildWrappers();
+      beforeEach(() => {
+        mockScopedSlots = { input: '<input v-bind="props.inputProps" />' };
+
+        updateWrapper();
       });
 
       it('should provide proper label prop to input element', () => {
-        expect(input.attributes('label')).toEqual(basePropsData.label);
+        expect(input.attributes('label')).toEqual(baseProps.label);
```

Comment:
IMO Is the same.

You can use `MOCK_PROP` and use it inside `baseProps` and then match `MOCK_PROP` in assertion or directly use `baseProps`.

I would preffer to use directly `baseProps`, because there is no need to use `MOCK_`.

But it depends on the test, maybe in some case you will prefer to `MOCK_` because you have a bunch of assertions.

The important thing is avoid use direct values on assertions if you.
If you will use direct values, make sure it will be the only one. If you need to use it in multiple tests, it will be better to have a `MOCK_` or some inmutable variable.

Let me show an example:

It is okay to do:
```
it(...){
expect(...).toBe(25);
}
it(...){
expect(...).toBe(26);
}
it(...){
expect(...).toBe(27);
}
```

It is NOT okay to do:
```
it(...){
expect(...).toBe(25);
}
it(...){
expect(...).toBe(25);
}
it(...){
expect(...).toBe(25);
}
```
In this case you should use `MOCK_` or some `base` prop